packages/next/client/index.tsx

```diff
@@ -647,10 +647,62 @@ if (process.env.__NEXT_RSC) {
   const {
     createFromFetch,
   } = require('next/dist/compiled/react-server-dom-webpack')
+
+  const encoder = new TextEncoder()
+  const serverDataBuffer = new Map<string, string[]>()
+  const serverDataWriter = new Map<string, WritableStreamDefaultWriter>()
+  const ssrCacheKey = location ? location.href : ''
+  function nextServerDataCallback(seg: [number, string, string]) {
```

Comment:
non-blocking, can do in later PR:
can give the name for `seg` properties in typing, like `(seg: [id: index, id: string, chunk: string])`